templates/dev.rake

```diff
@@ -1,4 +1,4 @@
-if Rails.env.development? || Rails.env.test?
+unless Rails.env.production?
```

Comment:
> Thanks for the discussion! I'll go ahead and change this PR.

Thank you too! :)

> Are you suggesting Staging uses the same DB instance as Production? Or were you referring to different types of databases?

I meant database types. In the servers we can configure the different database instances with `DATABASE_URL`, but we know how risky it is to configure it that way in development. Imagine if we had to set it to development after running tests... 😨 😆 

> So you're suggesting we add a different ENV variable for distinguishing staging from production, and code that depends on it? Any ideas what to call it? HEROKU_ENV?

Another env variable can get confusing indeed. Not sure how to even name it. In a project I added a Rails configuration:

```
--- a/config/application.rb
+++ b/config/application.rb
@@ -51,6 +51,10 @@ module PatientCheckIn

+    config.server_env = ENV.
+      fetch("APTIBLE_APPLICATION_HANDLE", Rails.env).
+      gsub("app-name-", "")
```

Don't love it either but it works.

What about adding many ENV variables? `FEATURE_A_ENABLED`, `HONEYBADGER_ENV`, `BUGSNAG`, etc? And we forget about almost-production Rails environments altogether? They can be properly named, and are easy to configure during tests too.

Still thinking!